src/FileType_XDMF.f90

```diff
@@ -0,0 +1,1236 @@
+  TYPE(StringType) :: str_name, str_value, str1, str2, toponame, xdmf_id_str 
+  INTEGER(SNK) :: nchildren, child_ctr
+  INTEGER(SLK) :: xdmf_id, nverts, ncells, i, j, vctr
+  CHARACTER(LEN=200) :: charpath
+  INTEGER(SLK), ALLOCATABLE :: vertex_list_2d(:, :), vertex_list_1d(:), cell_list_1d(:)
+
+  ! Create HDF5 group
+  CALL h5%mkdir(CHAR(mesh%name))
+
+  ! Determine number of xmle children
+  ! Geometry, Topology, Material, Cell sets
+  ! Assumes mesh has geometry and topology
+  nchildren = 2
+  IF(ALLOCATED(mesh%material_ids)) nchildren = nchildren + 1
+  IF(ALLOCATED(mesh%cell_sets)) nchildren = nchildren + SIZE(mesh%cell_sets)
+  ALLOCATE(xmle%children(nchildren))
+
+  ! GEOMETRY
+  current_xml => xmle%children(1)
+  str_name="Geometry"
+  CALL current_xml%setName(str_name) 
+  current_xml%nAttr=0
+  current_xml%parent => xmle
+
+  str_name= "GeometryType"
+  str_value = "XYZ"
+  CALL current_xml%setAttribute(str_name, str_value)
+
+  ALLOCATE(current_xml%children(1))
+  child_xml => current_xml%children(1)
+  str_name="DataItem"
+  CALL child_xml%setName(str_name) 
+  child_xml%nAttr=0
+  child_xml%parent => current_xml
+
+  str_name= "DataType"
+  str_value = "Float"
+  CALL child_xml%setAttribute(str_name, str_value)
+
+  nverts=SIZE(mesh%vertices, DIM=2)
+  str_name="Dimensions"
+  str1 = nverts
+  str2 = "3"
+  str_value = str1//" "//str2
+  CALL child_xml%setAttribute(str_name, str_value)
+
+  str_name= "Format"
+  str_value = "HDF"
+  CALL child_xml%setAttribute(str_name, str_value)
+
+  str_name= "Precision"
+  str_value = "8"
+  CALL child_xml%setAttribute(str_name, str_value)
+
+  i = LEN_TRIM(strpath)
+  charpath = CHAR(strpath)
+  child_xml%content = charpath(1:i-4)//"h5:/"//mesh%name//"/vertices"
+
+  CALL h5%fwrite(CHAR(mesh%name)//'->vertices',mesh%vertices)
+
+  ! TOPOLOGY
+  current_xml => xmle%children(2)
+  str_name="Topology"
+  CALL current_xml%setName(str_name) 
+  current_xml%nAttr=0
+  current_xml%parent => xmle
+
+  ! Single topology
+  IF(mesh%singleTopology)THEN
+    str_name= "TopologyType"
+    xdmf_id = mesh%cells(1)%vertex_list(1)
+    xdmf_id_str = xdmf_id
+    CALL XDMFTopologyList%get('XDMFID->'//ADJUSTL(xdmf_id_str), toponame)
+    CALL XDMFTopologyList%get(ADJUSTL(toponame)//'->n', nverts)
+    str_value = toponame
+    CALL current_xml%setAttribute(str_name, str_value)
+
+    str_name= "NumberOfElements"
+    ncells = SIZE(mesh%cells)
+    str_value = ncells
+    CALL current_xml%setAttribute(str_name, str_value)
+
+    str_name= "NodesPerElement"
+    str_value = nverts
+    CALL current_xml%setAttribute(str_name, str_value)    
+
+    ALLOCATE(current_xml%children(1))
+    child_xml => current_xml%children(1)
+    str_name="DataItem"
+    CALL child_xml%setName(str_name) 
+    child_xml%nAttr=0
+    child_xml%parent => current_xml
+
+    str_name= "DataType"
+    str_value = "Int"
+    CALL child_xml%setAttribute(str_name, str_value)
+
+    str_name="Dimensions"
+    str1 = ncells
+    str2 = nverts
```

Comment:
nverts is set to str1 at line 879. can set str2 to ncells instead of reassigning both.

Reply:
True, but 50 lines is a ways away and assignment is extremely fast, so I think this makes sense.